modules/apps/forms-and-workflow/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js

```diff
@@ -360,30 +360,71 @@ AUI.add(
 				_invokeResourceURL: function(params) {
 					var instance = this;
 
+					var payload;
+
 					var url = Liferay.PortletURL.createResourceURL();
 
-					url.setParameters(params.queryParameters);
-					url.setPortletId(instance.ID);
-					url.setResourceId(params.resourceId);
+					if (params.queryParameters && params.queryParameters.calendarIds && params.queryParameters.calendarIds.length > 50) {
+						var beginIndex = 0;
+						var endIndex = 50;
 
-					var payload;
+						var calendarIds = params.queryParameters.calendarIds.split(',');
 
-					if (params.payload) {
-						payload = Liferay.Util.ns(instance.get('namespace'), params.payload);
+						var length = calendarIds.length;
+
+						while (length > 0) {
+							params.queryParameters.calendarIds = calendarIds.slice(beginIndex, endIndex).join(',');
+
+							url.setParameters(params.queryParameters);
+							url.setPortletId(instance.ID);
+							url.setResourceId(params.resourceId);
+
+							if (params.payload) {
+								payload = Liferay.Util.ns(instance.get('namespace'), params.payload);
+							}
+
+							A.io.request(
+								url.toString(),
+								{
+									data: payload,
+									dataType: 'JSON',
+									on: {
+										success: function() {
+											params.callback(this.get('responseData'));
+										}
+									}
+								}
+							);
+
+							beginIndex = endIndex;
+
```

Comment:
@dustinryerson, another comment is regarding the use of the ternary operator. Do you think BChan will allow the use of it? Me and Adam are a bit unsure about his opinion.